packages/flutter/test/material/chip_theme_test.dart

```diff
@@ -200,14 +256,67 @@ void main() {
           data: chipTheme,
           child: Builder(
             builder: (BuildContext context) {
-              return Directionality(
-                textDirection: TextDirection.ltr,
-                child: Material(
-                  child: Center(
-                    child: RawChip(
-                      label: const SizedBox(width: 100, height: 100),
-                      onSelected: (bool newValue) { },
-                    ),
+              return const Material(
+                child: Center(
+                  child: Chip(
+                    label: SizedBox(width: 100, height: 100),
+                  ),
+                ),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    final RenderBox materialBox = getMaterialBox(tester);
+    expect(materialBox, paints..rect(color: chipTheme.backgroundColor));
+    expect(tester.getSize(find.byType(RawChip)), const Size(250, 250)); // icon + label + padding + labelPadding
+    expect(getMaterial(tester).elevation, chipTheme.elevation);
+    expect(getMaterial(tester).shape, chipTheme.shape);
+    expect(getLabelStyle(tester).style.fontSize, 32);
+  });
+
+  testWidgets('Chip with Avatar Icon custom iconTheme and uses ChipTheme', (WidgetTester tester) async {
+    const IconThemeData shadowedChipThemeIconTheme = IconThemeData(color: Color(0xffbbeeff));
+    const ChipThemeData chipTheme = ChipThemeData(
+      backgroundColor: Color(0xff112233),
+      elevation: 4,
+      padding: EdgeInsets.all(50),
+      labelPadding: EdgeInsets.all(25),
+      labelStyle: TextStyle(fontSize: 32),
+      shape: RoundedRectangleBorder(),
+      iconTheme: shadowedClipThemeIconTheme,
```

Comment:
well, i messed up a bit (not the first time while using GH to write code). fixing...

```suggestion
      iconTheme: shadowedChipThemeIconTheme,
```